x-pack/plugins/observability_solution/synthetics/server/alert_rules/status_rule/monitor_status_rule.ts

```diff
@@ -104,59 +80,103 @@ export const registerSyntheticsStatusCheckRule = (
         savedObjectsClient,
         scopedClusterClient.asCurrentUser,
         server,
-        syntheticsMonitorClient
+        syntheticsMonitorClient,
+        options
       );
 
-      const { downConfigs, staleDownConfigs, upConfigs } = await statusRule.getDownChecks(
-        ruleState.meta?.downConfigs as OverviewStatus['downConfigs']
-      );
+      const { downConfigs, staleDownConfigs, upConfigs, monitorLocationsMap } =
+        await statusRule.getDownChecks(ruleState.meta?.downConfigs as StatusConfigs);
+
+      const isCustomRule = !isEmpty(params);
 
-      Object.entries(downConfigs).forEach(([idWithLocation, { ping, configId }]) => {
-        const locationId = ping.observer.name ?? '';
-        const alertId = idWithLocation;
-        const monitorSummary = getMonitorSummary(
-          ping,
-          DOWN_LABEL,
-          locationId,
-          configId,
-          dateFormat,
-          tz
-        );
+      const { isLocationBased, downThreshold, numberOfLocations } = getConditionType(
+        params.condition
+      );
 
-        const { uuid, start } = alertsClient.report({
-          id: alertId,
-          actionGroup: MONITOR_STATUS.id,
+      const getConfigsByIds = () => {
```

Comment:
Can we move this function out of the executor function and reference it in the executor function.